src/docs/sdk/research/performance/index.mdx

```diff
@@ -192,6 +192,160 @@ In the example above, if an error bubbles up the call stack we want to be able t
+
+However, it was also clear that some form of lifecycle hook was necessary to allow users to do things like updating a transaction name.
+
+We ended up with the middle ground of allowing the mutation/dropping of transaction events through `EventProcessor` (a more general form of `BeforeSend`), which solves problems by giving users immediate access to their data before it goes out of the SDK, but is also the worse of both worlds in that it exposes the transaction duality that was never intended to leak and is more complicated to use than `BeforeSend`.
+
+In OpenTelemetry, spans go through span processors, which are two lifecycle hooks: one when a span is started and one when it is ended.
+
+### Nested Transactions
+
+On one hand, Sentry's ingestion model does not allow for nested transactions. Transactions are meant to mark service transitions.
+
+On the other hand, Sentry as a product requires that transactions exist. Data outside of a transaction is lost.
+
+In practice, SDKs have no way of preventing transactions from becoming nested. The end result is possibly surprising to users, as each transaction starts a new tree. The only way to relate those trees is through the `trace_id`.
+
+Sentry's billing model is per event/transaction. That means that a transaction within a transaction generates two billable events.
+
+In SDKs, a transaction within a transaction causes inner spans to be "swallowed" by the innermost transaction surrounding them. In some situations, automatic instrumentation creating spans will only make them appear in one of the two transactions, but not both, causing apparent gaps in instrumentation.
+
+The product does not know how to deal with nested transactions yet. If the transactions share a `trace_id` they can be visualized in the trace view. To get to the trace view, one must follow a link from one of the transactions. However, when looking at any one transaction it is as if all the others did not exist (either ancestor or descendent transactions are not represented on screen).
+
+There is also user confusion about what one would expect in the UI for such a situation (pseudocode):
+
+```python
+# if do_a_database_query returns 10 results, is the user
+#   - seeing 11 transactions in the UI?
+#   - billed for 11 transactions?
+#   - see spans within create_thumbnail in the innermost transaction only?
+with transaction("index-page"):
+    results = do_a_database_query()
+    for result in results:
+        if result["needs_thumbnail"]:
+            with transaction("create-thumbnail", {"resource": result["id"]}):
+                create_thumbnail(result)
+```
+
+### Spans Cannot Exist Outside of a Transaction
+
+Sentry's tracing experience is centered entirely around the part of a trace that exists inside transactions. This means that data cannot exist outside of a transaction even if it exists in a trace.
```

Comment:
A trace is the set of all spans that share a `trace_id`.

The current model forces us to have transactions (because they are the spans that can be transformed into events for ingestion), but they are not fundamental (and not part of other systems).

In theory, there can be traces with no transactions whatsoever.